### PR TITLE
[DOCS globals -> modules]

### PR DESCRIPTION
**Pull request by seanpdoyle**
Replace Global `App.` style documentation with `ember-cli` style ES6 modules
in files that follow the proper naming conventions.

### CODE REVIEW

File: packages/ember-application/lib/ext/controller.js [hunk 1]
```diff
@@ -83,7 +83,11 @@ ControllerMixin.reopen({
     For example, when you define a controller:
 
     ```javascript
-    App.CommentsController = Ember.ArrayController.extend({
+    // app/controllers/comments.js
+
+    import Ember from "ember";
+
+    export default Ember.ArrayController.extend({
```
Comment (seanpdoyle) [inline packages/ember-application/lib/ext/controller.js]: @trek Is this type of PR a good idea?

Too soon?

Not formatted properly?

If it's a good idea to move forward with this, I'll dig into other packages but I thought `ember-application` was a good enough package to get started in.

**Comment by trek** [comment]
@seanpdoyle we need to wait for 2.0 to do this. Technically globals are still the official default 1.x pattern, modules are possible only via ember-cli which is, as you say, still not a requirement. We also need to ship https://github.com/emberjs/website/issues/1855 so anyone on older versions can still see the 1.x series documentation.

**Comment by trek** [comment]
Part of the new docs will also be enabling the following pattern:

```app/routes/some-route.js
export default {};
```

which will detect that the language is `javascript` and the file location is `app/routes/some-route.js`. Before anyone "um actually"s me, The commonmark spec refers to the block after triple ticks only as "metadata" and allows any value there.

**Comment by fivetanley** [comment]
> Before anyone "um actually"s me

um actually it's called ECMAscript not javascript

**Comment by seanpdoyle** [comment]
:+1: @trek I'll join in when the time is right.

**Comment by trek** [comment]
> um actually it's called ECMAscript not javascript

um actually it's ECMAScript. Capital "s".

**Pull request closed without merging**